cognite/client/data_classes/contextualization.py

```diff
@@ -811,22 +871,42 @@ def _predictions_to_annotations(
         creating_app: str | None = None,
         creating_app_version: str | None = None,
     ) -> list[Annotation]:
-        return [
-            Annotation(
-                annotated_resource_id=item.file_id,
-                annotation_type=VISION_ANNOTATION_TYPE_MAP[prediction_type],
-                data=data,
-                annotated_resource_type="file",
-                status="suggested",
-                creating_app=creating_app or "cognite-sdk-python",
-                creating_app_version=creating_app_version or self._cognite_client.version,
-                creating_user=creating_user or None,
-            )
-            for item in self.items or []
-            if item.predictions is not None
-            for prediction_type, prediction_data_list in item.predictions.dump().items()
-            for data in prediction_data_list
-        ]
+        annotations = []
+
+        for item in self.items or []:
+            if item.predictions is not None:
+                for prediction_type, prediction_data_list in item.predictions.dump().items():
+                    for data in prediction_data_list:
+                        # check if multiple annotations represent the same prediction (e.g. in case of dial gauges, level gauges and valves)
+                        annotation_type = VISION_ANNOTATION_TYPE_MAP[prediction_type]
+                        if isinstance(annotation_type, dict):
+                            for key, value in annotation_type.items():
+                                annotation = Annotation(
+                                    annotated_resource_id=item.file_id,
+                                    annotation_type=value,
+                                    data=data[key],
+                                    annotated_resource_type="file",
+                                    status="suggested",
+                                    creating_app=creating_app or "cognite-sdk-python",
+                                    creating_app_version=creating_app_version or self._cognite_client.version,
+                                    creating_user=creating_user or None,
+                                )
+                                annotations.append(annotation)
+                        else:
```

Comment:
let's be explicit about the type here
```suggestion
elif isinstance(annotation_type, str):
  ...
else:
  raise TypeError("annotation_type must be str or dict")
```

Reply:
[19ba1af](https://github.com/cognitedata/cognite-sdk-python/pull/1414/commits/19ba1af3441bd761056a8644ce954fbf956da003)